app/public/src/Frame.js

```diff
@@ -0,0 +1,32 @@
+function Frame(roll1, roll2) {
+  this._rolls = [roll1, roll2]
+  this._MAX_PINS = 10
+};
+
+Frame.prototype.getRolls = function() {
+  return this._rolls;
+};
```

Comment:
In line 2, you declare `_rolls` as a private array. (a piece of state internal to the class). All your `Frame` methods control access to this array, which is great.

EXCEPT this one. This one just says "you know that private array I made earlier? It's not actually private. Any other class can use it. `Game` can use it. `Interface` can use it. Whatever"

This is super-confusing for me as a dev tasked with extending the code. On one hand, I understand that `Frame` should do all the business logic relating to managing an array of rolls. On the other hand, I'm being told from lines 6 to 8 that _any object_ can do the logic relating to managing the array of rolls.

The best bit is I know why you did this. You did it to make `Frame` testable (it's the only part of your code that uses this method). You wanted to test that frames contain rolls. Here's the secret – you don't need to test this. **Test behaviour, not state.** If the `Frame` object functions like it should, I don't care whether it contains rolls as an array, as a string, as an object, or as an AJAX call to some remote server sending out rolls.

Reply:
Hi Ed,
I think I get this but I am just trying to see what you mean by "use it", other classes can certainly see the array but not really use it right? I understand that they don't really have a need for it(as you correctly point out I don't use get Rolls in other classes...that said, I do use it in the following prototype method of a Frame:

Frame.prototype.getRollKnockedPins = function(index) {
  return this.getRolls()[index].getKnockedPins();
};

If I shouldn't be making a getter for this should I be accessing my rolls property directly then?

Thanks!